src/Controller/SalesChannelApiController.php

```diff
@@ -144,14 +153,20 @@ public function postPaymentDetails(
      * @Route(
      *     "/sales-channel-api/v1/adyen/payment-status",
      *     name="sales-channel-api.action.adyen.payment-status",
-     *     methods={"GET"}
+     *     methods={"POST"}
      * )
      *
+     * @param Request $request
      * @param SalesChannelContext $context
      * @return JsonResponse
      */
-    public function getPaymentStatus(SalesChannelContext $context): JsonResponse
+    public function getPaymentStatus(Request $request, SalesChannelContext $context): JsonResponse
     {
-        return new JsonResponse(['aloha']);
+        return new JsonResponse(
+            $this->paymentStatusService->getPaymentStatusWithOrderId(
+                $request->get('orderId'),
```

Comment:
please do a validation if the $request->get('orderId') is empty then return an error message

Reply:
Done